source/building-your-own-javascript-components/index.html.md.erb

```diff
@@ -211,14 +189,14 @@ To avoid reimplementing features shared across components, we recommend that you
 By default, `createAll` looks through the whole page for elements to initialise the components on. If you update a page with new markup, like a modal dialog box, you can initialise components on that part of the page only.
 
 ```js
-import {createAll} from ‘govuk-frontend’
-import {ProjectComponent} from ‘./project-component.js’
+import { createAll } from 'govuk-frontend'
+import { ProjectComponent } from './project-component.js'
 
-const $element = document.querySelector(‘.app-dialog’)
+const $element = document.querySelector('.app-dialog')
 
 createAll(ProjectComponent, {SOME_CONFIGURATION}, $element)
-// If the component does not need any configuration, pass `null` for the configuration
-createAll(ProjectComponent, null, $element)
+// If the component does not need any configuration, pass `undefined` for the configuration
+createAll(ProjectComponent, undefined, $element)
```

Comment:
Sorry @romaricpascal

I've just tried this code and when `document.querySelector('.app-dialog')` returns `null` it errors:

```console
TypeError: null is not an object (evaluating 'createAllOptions.scope')
```

Unfortunately this is because `null` passes the [`typeof createAllOptions === 'object'` check](https://github.com/alphagov/govuk-frontend/blob/3cbc709d489e04a468d9032a942f26870af656d2/packages/govuk-frontend/src/govuk/init.mjs#L78)

Or maybe the docs need fixing?

The JSDoc types `Element | Document` for `$scope` lack `null` so you'd need:

```suggestion
if ($element) {
  createAll(ProjectComponent, undefined, $element)
}
```

Reply:
Aware that checking for `null` would fall back to the default `$scope = document` unintentionally

```patch
- if (typeof createAllOptions === 'object') {
+ if (!!createAllOptions && typeof createAllOptions === 'object') {
```

Maybe widening **$scope** to `Element | Document | null` only and letting `onError` be called is better?

But the `onError` wrapper doesn't kick in for the options/scope checks

Reply:
Oh, cheers for letting us know. @patrickpatrickpatrick created [an issue on the `govuk-frontend` repository to track this](https://github.com/alphagov/govuk-frontend/issues/5418). Once we pick a way, we can update the docs accordingly.

Reply:
Yeah we had a nice chat on Slack, thanks @romaricpascal